frontend/javascripts/oxalis/model/accessors/dataset_accessor.js

```diff
@@ -515,6 +544,40 @@ export function getEnabledLayers(
   });
 }
 
+function _getMissingLayersForCurrentZoom(state: OxalisState) {
+  const { dataset } = state;
+  const zoomStep = getRequestLogZoomStep(state);
+
+  const { renderMissingDataBlack } = state.datasetConfiguration;
+  const maxZoomStepDiff = getMaxZoomStepDiff(state.datasetConfiguration.loadingStrategy);
+
+  const missingLayers = getEnabledLayers(dataset, state.datasetConfiguration)
+    .map((layer: DataLayerType) => ({
+      layer,
+      resolutionInfo: getResolutionInfo(layer.resolutions),
+    }))
+    .filter(({ resolutionInfo }) => {
+      const isMissing = !resolutionInfo.hasIndex(zoomStep);
+      if (!isMissing || renderMissingDataBlack) {
+        return isMissing;
+      }
```

Comment:
I'm having a little trouble wrapping my head around this, is it correct that only `isMissing` is returned without `renderMissingDataBlack` influencing the return value?

Reply:
I agree that this is kind of hard to read. I think I'll restructure the code a bit to make it more intuitive. Probably this:

```js
const isMissing = !resolutionInfo.hasIndex(zoomStep);
if (!isMissing) {
  // The layer exists. Thus, it is not missing.
  return false;
}

if (renderMissingDataBlack) {
  // We already know that the layer is missing. Since `renderMissingDataBlack`
  // is enabled, the fallback magnifications don't matter. The layer cannot be
  // rendered.
  return true;
}

...
```

Reply:
I'll also change `missing` to `unrenderable` since this fits better in my opinion.